sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -119,41 +119,45 @@ export async function deserializeAtomXmlResponse(
       const HttpResponseCodes: any = Constants.HttpResponseCodes;
       const statusCode = response.status;
       if (isKnownResponseCode(statusCode)) {
-        throw buildError(
-          {
-            code: HttpResponseCodes[statusCode]
-          },
-          response
+        throw new RestError(
+          "Service Error",
+          HttpResponseCodes[statusCode],
+          response.status,
+          stripRequest(response.request),
+          stripResponse(response)
         );
       } else {
-        throw buildError(
-          {
-            code: `UnrecognizedHttpResponseStatus: ${statusCode}`
-          },
-          response
+        throw new RestError(
+          "Service Error",
+          `UnrecognizedHttpResponseStatus: ${statusCode}`,
+          response.status,
+          stripRequest(response.request),
+          stripResponse(response)
         );
       }
     } else {
       throw buildError(errorBody, response);
     }
   }
 
-  response.parsedBody = parseAtomResult(response.parsedBody, nameProperties);
+  response.parsedBody = parseAtomResult(response, nameProperties);
```

Comment:
In such a case where the method takes an object and returns a value to be updated on the same object, I ideally the method to do the update internally.

@bterlson Do you have a preference in such cases?

Reply:
Updated